spec/parsers/exif_parser_spec.rb

```diff
@@ -1,15 +1,88 @@
-        expect(filename).to include(result.orientation_sym.to_s)
+  describe 'EXIFParser#exif_from_tiff_io' do
+    describe 'Orientation' do
+      describe 'is able to correctly parse orientation for all the TIFF EXIF examples from FastImage' do
+        Dir.glob(fixtures_dir + '/exif-orientation-testimages/tiff-*/*.tif').each do |tiff_path|
+          filename = File.basename(tiff_path)
+          it "is able to parse #{filename}" do
+            result = described_class.exif_from_tiff_io(File.open(tiff_path, 'rb'))
+            expect(result).not_to be_nil
+            expect(result.orientation_sym).to be_kind_of(Symbol)
+            # Filenames in this dir correspond with the orientation of the file
+            expect(filename).to include(result.orientation_sym.to_s)
+          end
+        end
+      end
+
+      it 'is able to deal with an orientation tag which a tuple value for orientation' do
+        path = fixtures_dir + '/EXIF/double_orientation.exif.bin'
+        exif_data = File.open(path, 'rb') do |f|
+          described_class.exif_from_tiff_io(f)
+        end
+        expect(exif_data.orientation).to eq(1)
+      end
+    end
+
+    describe 'SubIFDs' do
+      it 'should not retrieve subIFDs data by default' do
+        path = fixtures_dir + 'NEF/RAW_NIKON_D40_SRGB.NEF'
+
+        exif_data = File.open(path, 'rb') do |f|
+          described_class.exif_from_tiff_io(f)
+        end
+
+        expect(exif_data.sub_ifds_data).not_to be_nil
+        expect(exif_data.sub_ifds_data).to eq({})
+      end
+
+      it 'is able retrieve data from all subIFDs optionally' do
+        # Verifying:
+        # {
+        #    offset_1 => { subIFD_1 data...}
+        #    offset_2 => { subIFD_2 data...}
+        # }
+
+        path = fixtures_dir + 'NEF/RAW_NIKON_D40_SRGB.NEF'
+        should_include_sub_ifds = true
+
+        exif_data = File.open(path, 'rb') do |f|
+          described_class.exif_from_tiff_io(f, should_include_sub_ifds)
+        end
+
+        offset_1, offset_2 = exif_data.sub_ifds
+        sub_ifds_data = exif_data.sub_ifds_data
+
+        expect(sub_ifds_data).not_to be_nil
+        expect(sub_ifds_data).to have_key(offset_1)
+        expect(sub_ifds_data).to have_key(offset_2)
+      end
+
+      it 'returns EXIFR IFD instances as subIFD data' do
+        # Verifying:
+        # {
+        #    offset_1 => { new_subfile_type => 1, ...}
+        #    offset_2 => { new_subfile_type => 0, ...}
+        # }
+        # we shouldn't verify everything, since we trust to EXIFR for that.
+        # making sure we are getting each subfile type should be good enough.
+
+        path = fixtures_dir + 'NEF/RAW_NIKON_D40_SRGB.NEF'
+        should_include_sub_ifds = true
+
+        exif_data = File.open(path, 'rb') do |f|
+          described_class.exif_from_tiff_io(f, should_include_sub_ifds)
+        end
+
+        offset_1, offset_2 = exif_data.sub_ifds.sort
+        first_sub_ifd = exif_data.sub_ifds_data&.[](offset_1)
```

Comment:
If we have the actual file and thus know its EXIF data, why do we still have to guard ourselves with `&.`?

Reply:
This tries to make makes sure that, if we mess and break `exif_from_tiff_io` in the future, the errors will be raised by the assertions and not by this line -- making it easier to follow at first glance and better to maintain.

Reply:
I see your point, but on those grounds, anything could break in the future and I'd argue broken code should surface from expectations and not just...break 😊

Reply:
Alright, got it 👍